src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/CompatibilitySuppressions.xml

```diff
@@ -0,0 +1,82 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<!-- https://learn.microsoft.com/en-us/dotnet/fundamentals/package-validation/diagnostic-ids -->
+<Suppressions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+  <Suppression>
+    <DiagnosticId>CP0002</DiagnosticId>
+    <Target>M:Microsoft.Extensions.DependencyInjection.ObjectFactory.#ctor(System.Object,System.IntPtr)</Target>
+    <Left>lib/net6.0/Microsoft.Extensions.DependencyInjection.Abstractions.dll</Left>
+    <Right>lib/net7.0/Microsoft.Extensions.DependencyInjection.Abstractions.dll</Right>
+  </Suppression>
+  <Suppression>
+    <DiagnosticId>CP0002</DiagnosticId>
+    <Target>M:Microsoft.Extensions.DependencyInjection.ServiceDescriptor.get_IsKeyedService</Target>
+    <Left>ref/net462/Microsoft.Extensions.DependencyInjection.Abstractions.dll</Left>
+    <Right>lib/net462/Microsoft.Extensions.DependencyInjection.Abstractions.dll</Right>
+  </Suppression>
+  <Suppression>
+    <DiagnosticId>CP0002</DiagnosticId>
+    <Target>M:Microsoft.Extensions.DependencyInjection.ServiceDescriptor.get_KeyedServiceType</Target>
+    <Left>ref/net462/Microsoft.Extensions.DependencyInjection.Abstractions.dll</Left>
+    <Right>lib/net462/Microsoft.Extensions.DependencyInjection.Abstractions.dll</Right>
+  </Suppression>
+  <Suppression>
+    <DiagnosticId>CP0002</DiagnosticId>
+    <Target>M:Microsoft.Extensions.DependencyInjection.ServiceDescriptor.get_IsKeyedService</Target>
+    <Left>ref/net6.0/Microsoft.Extensions.DependencyInjection.Abstractions.dll</Left>
+    <Right>lib/net6.0/Microsoft.Extensions.DependencyInjection.Abstractions.dll</Right>
+  </Suppression>
+  <Suppression>
+    <DiagnosticId>CP0002</DiagnosticId>
+    <Target>M:Microsoft.Extensions.DependencyInjection.ServiceDescriptor.get_KeyedServiceType</Target>
+    <Left>ref/net6.0/Microsoft.Extensions.DependencyInjection.Abstractions.dll</Left>
+    <Right>lib/net6.0/Microsoft.Extensions.DependencyInjection.Abstractions.dll</Right>
+  </Suppression>
+  <Suppression>
+    <DiagnosticId>CP0002</DiagnosticId>
+    <Target>M:Microsoft.Extensions.DependencyInjection.ServiceDescriptor.get_IsKeyedService</Target>
+    <Left>ref/net7.0/Microsoft.Extensions.DependencyInjection.Abstractions.dll</Left>
+    <Right>lib/net7.0/Microsoft.Extensions.DependencyInjection.Abstractions.dll</Right>
+  </Suppression>
+  <Suppression>
+    <DiagnosticId>CP0002</DiagnosticId>
+    <Target>M:Microsoft.Extensions.DependencyInjection.ServiceDescriptor.get_KeyedServiceType</Target>
+    <Left>ref/net7.0/Microsoft.Extensions.DependencyInjection.Abstractions.dll</Left>
+    <Right>lib/net7.0/Microsoft.Extensions.DependencyInjection.Abstractions.dll</Right>
+  </Suppression>
+  <Suppression>
+    <DiagnosticId>CP0002</DiagnosticId>
+    <Target>M:Microsoft.Extensions.DependencyInjection.ServiceDescriptor.get_IsKeyedService</Target>
```

Comment:
@ViktorHofer PTAL at these suppressions here - these are new APIs so should we conditionally exclude them in the .cs \ .csproj to avoid these suppressions?

Reply:
These suppressions mean that the API is either missing from the contract or the implementation. They must be in sync.

Reply:
So should I remove these suppressions?

Reply:
If the build is green without them, then yes, please remove them. Why were they necessary in the first place?

Reply:
I just followed the doc from here: https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/updating-ref-source.md